corehq/apps/api/resources/auth.py

```diff
@@ -53,12 +53,14 @@ def __init__(self, allow_session_auth=False, *args, **kwargs):
                 'digest': login_or_digest,
                 'basic': login_or_basic,
                 'api_key': login_or_api_key,
+                'oauth2': login_or_oauth2,
             }
         else:
             self.decorator_map = {
                 'digest': digest_auth,
                 'basic': basic_auth,
                 'api_key': api_key_auth,
+                'oauth2': oauth2_auth,
```

Comment:
these two lines are what enable oauth access for any API that uses or extends `LoginAndDomainAuthentication`.

Reply:
Do you think this should also be applied to 
https://github.com/dimagi/commcare-hq/blob/a9046de9fc4ad582d944b296b2ed8d9cfb56c82f/corehq/apps/domain/decorators.py#L310-L312

Reply:
Hmm, good question. On the one hand if the intention of that decorator is to only be used by APIs then I think it makes sense. On the other hand, limiting the surface area/footprint of the permissions to just the tastypie resources makes me feel a little more comfortable about any potential unintended side effects.

One thing I'm actually quite confused about but haven't tested - based on my other comment above, where do the domain checks happen for functions decorated with `api_auth`? Most have other permission checks, but I found two that appear to not have any other checks:

https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/export/views/list.py#L759-L763
https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/views/cli.py#L27-L29

Reply:
nevermind, `_login_or_challenge` handles that. :sweat_smile:

Reply:
anyway, back to your original question - I don't have a strong opinion. happy to add it if you think it makes sense?

Reply:
I like the idea of having a one-stop-shop for API auth, so we know what's considered a public API, and so API access is somewhat consistent.  However we do already have a split between tastypie and other APIs, so I think it's fine to punt in the hopes that we eventually unify those two.

Reply:
Hmm, that seems logical to me. I'll make a todo to add it in a follow up.

> in the hopes that we eventually unify those two.